core/src/saros/negotiation/additional_resource_data/AbstractPossibleRepresentationProvider.java

```diff
@@ -0,0 +1,138 @@
+   * object and a path of the reference point resource relative to the reference object.
+   *
+   * <p>The paths contained in the mapping must always use Unix separators.
+   *
+   * @param referencePoint the reference point
+   * @return a list of possible local representations of the reference point resource
+   */
+  protected abstract List<Pair<String, String>> computePossibleRepresentations(
+      IReferencePoint referencePoint);
+
+  @Override
+  public Map<String, String> getMapping(IReferencePoint referencePoint) {
+    List<Pair<String, String>> possibleRepresentations =
+        computePossibleRepresentations(referencePoint);
+
+    String flattenedPossibleRepresentations =
+        possibleRepresentations
+            .stream()
+            .map(
+                pair ->
+                    ESCAPE_INNER.escape(pair.getLeft())
+                        + DELIMITER_INNER
+                        + ESCAPE_INNER.escape(pair.getRight()))
+            .map(ESCAPE_OUTER::escape)
+            .collect(Collectors.joining(DELIMITER_OUTER));
+
+    return Collections.singletonMap(POSSIBLE_REPRESENTATIONS_KEY, flattenedPossibleRepresentations);
+  }
+
+  /**
+   * Returns the possible representations contained in the given resource negotiation data.
+   *
+   * <p>The paths contained in the mapping always use Unix separators.
+   *
+   * @param resourceNegotiationData the resource negotiation data
+   * @return the possible representations contained in the given resource negotiation data or an
+   *     empty list of no such representations are specified by the negotiation data
+   */
+  public static List<Pair<String, String>> getPossibleRepresentations(
+      ResourceNegotiationData resourceNegotiationData) {
+    String flattenedPossibleRepresentation =
+        resourceNegotiationData.getAdditionalResourceData().get(POSSIBLE_REPRESENTATIONS_KEY);
+
+    if (flattenedPossibleRepresentation == null) {
+      return Collections.emptyList();
+    }
+
+    return split(flattenedPossibleRepresentation);
+  }
+
+  /**
+   * Splits and unescapes given flattened possible representation string. Returns the possible
+   * representations contained in the given string.
+   *
+   * @param flattenedPossibleRepresentations the flattened possible representation string contained
+   *     in the resource negotiation data
+   * @return the possible representations contained in the given string
+   */
+  private static List<Pair<String, String>> split(String flattenedPossibleRepresentations) {
+    return ESCAPE_OUTER
+        .getSplitPattern()
+        .splitAsStream(flattenedPossibleRepresentations)
+        .map(ESCAPE_OUTER::unescape)
+        .map(
+            joinedPossibleRepresentation -> {
+              String[] values = ESCAPE_INNER.getSplitPattern().split(joinedPossibleRepresentation);
+
+              if (values.length != 1 && values.length != 2) {
+                throw new IllegalStateException(
```

Comment:
I guess this exception is not catched anywhere and I guess this is always called from some UI code. In other words bad data can make the UI non useable which is not very good. Isn't is better just to discard the entries or simply not return anything ?

Reply:
Done. Replaced with error log.